src/REmatchEngine.cpp

```diff
@@ -15,6 +15,8 @@ const char NFA_NSTATES_NAME[] = "nstates";
 
 REmatchAutomataEngine::REmatchAutomataEngine()
     : flow(nullptr), matcher(nullptr), txtbl(nullptr) {}
+REmatchAutomataEngine::REmatchAutomataEngine(bool red)
+    : flow(nullptr), matcher(nullptr), txtbl(nullptr), reduce(red) {}
```

Comment:
Use delegating constructor to avoid redundancy.
``` c++
REmatchAutomataEngine::REmatchAutomataEngine(bool red)
    : REmatchAutomataEngine(), reduce(red) {}
```